app/src/main/java/unipd/se18/ocrcamera/ImageProcessing.java

```diff
@@ -0,0 +1,349 @@
+        Mat img = Imgcodecs.imread(imagePath);
+
+        //Call of internal methods
+        RotatedRect area = detectMaxTextArea(imagePath);
+        img = crop(area, img);
+        Bitmap image = conversion(img);
+        return image;
+    }
+
+
+    /**
+     * Calculate the angle between the text and the horizontal
+     * @param imagePath path of the image you want to analyze
+     * @return the angle between the text and the horizontal
+     * @throws FileNotFoundException if imagePath doesn't exist
+     * @author Thomas Porro (g1)
+     */
+    public double computeSkew(String imagePath) throws FileNotFoundException {
+
+        //Turns the image in grayscale and put it in a matrix
+        Log.d(TAG, "Image path = " + imagePath);
+        Mat img = conversion(imagePath);
+        //save(img, "grayScale.jpg");
+
+        //Invert the colors of "img" onto itself
+        Core.bitwise_not(img, img);
+
+        //Detect the edges in the image
+        double threshold1 = 50;
+        double threshold2 = 200;
+        int apertureSize = 3;
+        boolean l2gradient = false;
+        Imgproc.Canny(img, img, threshold1, threshold2, apertureSize, l2gradient);
+
+        //Create a 4 dimensions vector using matrix
+        MatOfInt4 lines = new MatOfInt4();
+
+        //Process the image with the Probabilistic Hough Transform
+        double rho = 1;
+        double theta = Math.PI / 180;
+        int threshold = 50;
+        double minLineLenght = 50;
+        double maxLineGap = 10;
+        Imgproc.HoughLinesP(img, lines, rho, theta, threshold, minLineLenght, maxLineGap);
+
+        double meanAngle = 0;
+        Log.d(TAG, "rows = " + lines.cols() + "\ncols = " + lines.cols());
+
+        //Analizes the text line per line
+        for (int i = 0; i < lines.rows(); i++) {
+            //Get points from the beginning and the ending of the line of text
+            double[] vec = lines.get(i, 0);
+
+            //First point
+            double x1 = vec[0];
+            double y1 = vec[1];
+
+            //Second point
+            double x2 = vec[2];
+            double y2 = vec[3];
+
+            //Sum all the angle of the lines of text
+            meanAngle += Math.atan2(y2 - y1, x2 - x1);
+        }
+
+        //Calculate the meanAngle by dividing it with the number of rows
+        meanAngle /= lines.rows();
+
+        //Transform the angle in degrees
+        double degreesAngle = Math.toDegrees(meanAngle);
+        Log.i(TAG, "Mean angle=" + degreesAngle);
+        return degreesAngle;
+    }
+
+
+    /**
+     * Detect in which region of the picture there is some text
+     * @param imagePath the path of the image you want to analyze
+     * @return the rectangle which contains the text
+     * @throws FileNotFoundException if imagePath doesn't exist
+     * @author Thomas Porro (g1), Oscar Garrido (g1)
+     */
+    public RotatedRect detectMaxTextArea(String imagePath) throws FileNotFoundException {
```

Comment:
Is this a method to find the text with the biggest area on the pic? Or does this method find a region with some text?

Reply:
ok, thanks, could be a good idea splitting this method in more than one unique method? Anyway it's resolved.